packages/openapi-generator/src/parser/schema.ts

```diff
@@ -160,7 +174,37 @@ function parseObjectSchemaProperties(
           ? undefined
           : propSchema.description,
         name: propName,
-        required: schema.required?.includes(propName) || false
+        required: schema.required?.includes(propName) || false,
+        typeProperties: {
+          format: isReferenceObject(propSchema) ? undefined : propSchema.format,
```

Comment:
I wonder if this can be done in a more generic way so that we do not have to do all the properties @marikaner